### PR TITLE
Upgrade add-on to Ember.js v5.8.1

### LINKED ISSUE #172: Upgrade Node.js to version >= 18
See: #171

Everything is related, #171 and https://github.com/DazzlingFugu/.github/pull/2 had to be created due to an issue caused by the old version of Node.js we currently: version 14 .

More details in https://github.com/DazzlingFugu/.github/pull/2

### PR DESCRIPTION
**Pull request by MrChocolatine**
#### Upgrade add-on to `ember-cli` v5.8.1 (#190)

The new requirements are:
- Ember.js v4.12 or above
- Ember CLI v4.12 or above
- Node.js v18 or above

---

Resolves #172 .

Closes #171 .
Closes #174 .
Closes #187 .
Closes #188 .
Closes #189 .

### CODE REVIEW

File: tests/dummy/config/ember-try.js [hunk 1]
```diff
@@ -47,38 +47,8 @@ module.exports = async function () {
...
-          },
-        },
-      }),
-      embroiderOptimized({
-        npm: {
-          dependencies: {
-            'stream-browserify': '3',
-          },
-        },
-      }),
```
Comment (MrChocolatine) [inline tests/dummy/config/ember-try.js:81]: Je vois que les tests (normaux + `ember-try`) passent sans ces objets de configuration pour `stream-browserify`...  Je n'ai pas cherché à savoir pourquoi ils ne sont plus nécessaires mais bon.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -17,15 +17,14 @@ jobs:
   org:
     uses: DazzlingFugu/.github/.github/workflows/js--emberjs-addons.yml@master
     with:
-      node-version: 14
+      node-version: 18
       package-manager: yarn
       ember-try-scenarios: "[
-          'ember-lts-3.28',
```
Comment (BlueCutOfficial) [inline .github/workflows/ci.yml:23]: Is there a specific change that makes the addon incompatible with Ember 3.28?

Comment (MrChocolatine) [reply .github/workflows/ci.yml:23]: It is in the blueprint:
https://github.com/ember-cli/ember-addon-output/blob/v5.8.1/.github/workflows/ci.yml#L60-L66

File: package.json [hunk 3]
```diff
@@ -42,56 +48,61 @@
...
+    "stylelint-config-standard": "^34.0.0",
+    "stylelint-prettier": "^4.1.0"
   },
   "peerDependencies": {
-    "ember-source": "^3.28.0 || ^4.0.0"
+    "ember-source": ">= 4.0.0"
   },
   "engines": {
-    "node": "14.* || 16.* || >= 18"
+    "node": ">= 18"
```
Comment (BlueCutOfficial) [inline package.json:105]: Same question for Node, but even more important. I would try to run the CI with the smallest possible version to keep the compatibility. If we need to remove 14 or/and 16, then this should be tracked as a breaking change.

File: tests/dummy/app/templates/application.hbs [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {{page-title "ember-reading-time"}}
 
-<h1>Welcome to ember-reading-time</h1>
+<h2 id="title">Welcome to ember-reading-time</h2>
```
Comment (BlueCutOfficial) [inline tests/dummy/app/templates/application.hbs:3]: Why this change? I am not aware of a `<h1>` already present in the page 🤔

File: .github/workflows/ci.yml (hunk 2 above)
Comment (MrChocolatine) [reply .github/workflows/ci.yml:23]: Bad news:	
https://github.com/DazzlingFugu/ember-reading-time/actions/runs/9102042426/job/25020773183

Comment (BlueCutOfficial) [reply .github/workflows/ci.yml:23]: Ok, thanks! it was worth it to give it a try :)

File: tests/dummy/app/templates/application.hbs (hunk 4 above)
Comment (MrChocolatine) [reply tests/dummy/app/templates/application.hbs:3]: Sorry I don't get what you mean. I am trying to understand.

Right now we have this:

https://github.com/DazzlingFugu/ember-reading-time/blob/d5e881dfff33cb3ce438a2e108631f01437744d1/tests/dummy/app/templates/application.hbs#L3

and the blueprint applies this:

https://github.com/ember-cli/ember-addon-output/blob/v5.8.1/tests/dummy/app/templates/application.hbs#L3

Why not just keeping what is generated by default?

Comment (BlueCutOfficial) [reply tests/dummy/app/templates/application.hbs:3]: My opinion is this change is unnecessary. There's no particular reason to replace our `<h1>` with a `<h2>`, it's just the main title of our page. I don't know why the blueprint has this content, but we have no reason to follow it here, this is our dummy app.

Comment (MrChocolatine) [reply tests/dummy/app/templates/application.hbs:3]: > but we have no reason to follow it here

Just to ease the upgrades by having less code diff.

Comment (MrChocolatine) [reply tests/dummy/app/templates/application.hbs:3]: I will revert that change, that's fine it's not blocking at all.
It's just, in my mind, while I understand what you said, I don't get why we want to deviate from the blueprint with such an unneeded change.

Comment (BlueCutOfficial) [reply tests/dummy/app/templates/application.hbs:3]: My points are: 
- the blueprint looks wrong. 
- the `application.hbs` is just a "starter", once you deviate from it you don't have any reason to change yours when you re-run the blueprint.

**Comment by MrChocolatine** [comment]
Can we move forward or I close this PR.

**Review by BlueCutOfficial — APPROVED** [review]
I was simply waiting for the revert I requested, the rest was all good to me.
